static/index.html

```diff
@@ -87,34 +85,30 @@
 			document.getElementById("clock").innerHTML = hour + colon + min + amOrPm;
 		}
 		
-		function readSpeed() {
+		function readSensor(endpoint, callback) {
 			var xhttp = new XMLHttpRequest();
 			xhttp.onreadystatechange = function() {
 				if (this.readyState == 4 && this.status == 200) {
-					var data = JSON.parse(this.responseText);
-					speedometer.set(data.speed);
+					callback(this.responseText);
 				} else {
 					// Display error to driver?
 				}
 			};
-			xhttp.open("GET", SPEED_ENDPOINT + "?" + Date.now(), true); // Date.now() prevents caching
+			xhttp.open("GET", endpoint + "?" + Date.now(), true); // Date.now() prevents caching
 			xhttp.send();
 		}
 		
-		function readSensors() { // Everything but speed, runs less frequently
-			var xhttp = new XMLHttpRequest();
-			xhttp.onreadystatechange = function() {
-				if (this.readyState == 4 && this.status == 200) {
-					var data = JSON.parse(this.responseText);
-					document.getElementById("battery-percent").innerText = data.battery_percent;
-					document.getElementById("error").style.display = data.bps_error ? "block" : "none";
-				} else {
-					// Display error to driver?
-				}
-			};
-			xhttp.open("GET", SENSORS_ENDPOINT + "?" + Date.now(), true); // Date.now() prevents caching
-			xhttp.send();
+		function updateSpeed() {
+			readSensor("/get-speed", speed => speedometer.set(speed));
+		}
+		function updateBatteryPercent() {
+			readSensor("/get-battery-percent", percent => document.getElementById("battery-percent").innerText = percent);
```

Comment:
[nit] This is pretty messy for the arrow function without brackets, but it's fine.